test/benchmark/l4bench.go

```diff
@@ -0,0 +1,194 @@
+package benchmarktest
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/cilium/cilium/pkg/logging"
+	. "github.com/cilium/cilium/test/ginkgo-ext"
+	"github.com/cilium/cilium/test/helpers"
+
+	. "github.com/onsi/gomega"
+	"github.com/sirupsen/logrus"
+)
+
+var _ = Describe("BenchmarkNetperfPerformance", func() {
+	var (
+		vm          *helpers.SSHMeta
+		monitorStop = func() error { return nil }
+
+		log    = logging.DefaultLogger
+		logger = logrus.NewEntry(log)
+
+		PerfLogFile   = "l4bench_perf.log"
+		PerfLogWriter bytes.Buffer
+	)
+
+	BeforeAll(func() {
+		vm = helpers.InitRuntimeHelper(helpers.Runtime, logger)
+		helpers.ExpectCiliumReady(vm)
+	})
+
+	AfterAll(func() {
+	})
+
+	JustBeforeEach(func() {
+		monitorStop = vm.MonitorStart()
+	})
+
+	JustAfterEach(func() {
+		vm.ValidateNoErrorsInLogs(CurrentGinkgoTestDescription().Duration)
+		Expect(monitorStop()).To(BeNil(), "cannot stop monitor command")
+	})
+
+	AfterFailed(func() {
+		vm.ReportFailed(
+			"cilium service list",
+			"cilium policy get")
+	})
+
+	AfterEach(func() {
+		LogPerm := os.FileMode(0666)
+		testPath, err := helpers.CreateReportDirectory()
+		Expect(err).Should(BeNil(), "cannot create log file")
+		helpers.WriteOrAppendToFile(filepath.Join(testPath, PerfLogFile), PerfLogWriter.Bytes(), LogPerm)
+		PerfLogWriter.Reset()
+	}, 500)
+
+	createContainers := func() {
+		vm.ContainerCreate(helpers.Client, helpers.NetperfImage, helpers.CiliumDockerNetwork, "-l id.client")
+		vm.ContainerCreate(helpers.Server, helpers.NetperfImage, helpers.CiliumDockerNetwork, "-l id.server")
+		vm.PolicyDelAll()
+		vm.WaitEndpointsReady()
+		err := helpers.WithTimeout(func() bool {
+			if data, _ := vm.GetEndpointsNames(); len(data) < 2 {
+				logger.Info("Waiting for endpoints to be ready")
+				return false
+			}
+			return true
+		}, "Endpoints are not ready", &helpers.TimeoutConfig{Timeout: 150})
+		Expect(err).Should(BeNil())
```

Comment:
Add a string as a second parameter to `Should`, e.g. `Should(BeNil(), fmt.Sprintf("timed out waiting for endpoints waiting to be ready while creating containers")`